File: coolrmi/hu.qgears.coolrmi/src/hu/qgears/coolrmi/messages/CoolRMIDisposeProxy.java
package hu.qgears.coolrmi.messages;

import java.io.Serializable;

import hu.qgears.coolrmi.serializer.PortableSerializable;


public class CoolRMIDisposeProxy
	extends AbstractCoolRMIMessage
	implements Serializable, PortableSerializable {
	private static final long serialVersionUID = 1L;
	private long proxyId;
	public long getProxyId() {
		return proxyId;
	}

	public CoolRMIDisposeProxy() {
		
	}

	public CoolRMIDisposeProxy(long queryId, 
			long proxyId) {
		super(queryId);
		this.proxyId = proxyId;
	}

	@Override
	public String toString() {
		return "Dispose proxy "+proxyId;
	}
	@Override
	public String getName() {
		return toString();
	}
}

File: coolrmi/CoolRMI.Net/Messages/CoolRMIDisposeProxy.cs
using System;
using CoolRMI.Net.Serializer;

namespace CoolRMI.Net.Messages
{
    [Serializable]
    public class CoolRMIDisposeProxy : AbstractCoolRMIMessage
    {
        [PortableFieldName("proxyId")]
        public long ProxyId { get; }
        public override string Name => ToString();

        public CoolRMIDisposeProxy(long queryId, long proxyId) : base(queryId)
        {
            ProxyId = proxyId;
        }

        public override string ToString()
        {
            return "Dispose proxy " + ProxyId;
        }
    }
}
